Browser WSS testing

MarcoPolo opened the pull request:
Browsers can only dial secure websocket addresses. So this also changes one Go implementation to listen on a TLS+WS (WSS) address.

Blocked on:
- [x] https://github.com/multiformats/js-mafmt/pull/141
- [x] https://github.com/libp2p/js-libp2p-websockets/pull/218

BigLep commented:
@MarcoPolo : is this unblocked now so can be moved out of draft and finished up?

MarcoPolo commented:
Yup! I'll circle back to this today

MarcoPolo commented:
This is ready for review. I changed the v0.24 Go implementation because it was easy and added a new js-libp2p v0.42 implementation with wss. After we merge this I'll backport the changes to other versions.

@marten-seemann could you review the Go changes (should be quick).
